test/functional/lua/version_spec.lua

```diff
@@ -0,0 +1,571 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear = helpers.clear
+local eq = helpers.eq
+local exec_lua = helpers.exec_lua
+local matches = helpers.matches
+local pcall_err = helpers.pcall_err
+
+local version = require('vim.version')
+
+local function quote_empty(s)
+  return tostring(s) == '' and '""' or tostring(s)
+end
+
+describe('version', function()
+  it('package', function()
+    clear()
+    eq({ major = 42, minor = 3, patch = 99 }, exec_lua("return vim.version.parse('v42.3.99')"))
```

Comment:
This tests the correctness of `runtime/lua/vim/_init_packages.lua` because `clear()` starts a new child Nvim process and `exec_lua` tries to use `vim.version`.

Previously, this went unnoticed because the rest of the tests use `require('vim.version')` (line 8) in the test runner process.